### PR TITLE
MM-58776 - Change Ancillary Permissions API to POST

### PR DESCRIPTION
**Pull request by sbishel**
#### Summary
When updating roles, the ancillary permissions are retrieved.  As part of a security issue, we have limited the length of the URL https://github.com/mattermost/mattermost/pull/26148. This limit causes issues with the `/ancillary` request, as it lists all necessary permissions as part of the query parameters.

This PR create a new POST endpoint for the `/ancillary` endpoint and updates the WebApp to use that endpoint instead. The GET endpoint is left for backwards compatibility.

Existing Unit tests now call the POST, as client4.go was updated to call the POST.

#### Ticket Link

Fixes https://mattermost.atlassian.net/browse/MM-58776

#### Release Note

```release-note
Fixed issue when saving custom roles.
```

### CODE REVIEW

**Review by enahum — APPROVED** [review]

File: server/channels/api4/permission.go
Comment (hanzei) [inline server/channels/api4/permission.go:16]: Is there a follow-up ticket for removing the `GET` endpoint that we can link in the code?

**Review by harshilsharma63 — APPROVED** [review]

File: server/channels/api4/permission.go
Comment (sbishel) [reply server/channels/api4/permission.go:16]: PR is created...will merge once v10 opens for merging. 
https://github.com/mattermost/mattermost/pull/27605

**Review by lindy65 — APPROVED** [review]
Thanks @sbishel - tested as per steps in Jira ticket and this is fixed/working as expected

**Pull request merged**
